### PR TITLE
Fixing ComboBox handlers and accessibility object initialization

### LINKED ISSUE #2432: ComboBox initialization/reinitialization fails with NullReferenceException
* .NET Core Version: 3.0
* Have you experienced this same bug with .NET Framework?: Yes

**Problem description:**
The issue is that the multi-form desktop application or application with complex control appearance behavior may fail with the error caused by NullReference which has ComboBox's origin. ComboBox initializes successfully and the issue does not reproduce in case this is shown initially on the forms and is not recreated while app lifecycle.

**Actual behavior:** 
`NullReferenceException` and application crash:
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Windows.Forms.ComboBox.WmParentNotify(Message& m)
   at System.Windows.Forms.ComboBox.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

**Expected behavior:**
No error.

**Minimal repro:**
Create the app with hidden dialog form and ComboBox on this form.
Start the app, trigger dialog form initialization and reinitialization without dialog showing.
Observe the error.

DevDiv issue: https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1022943

### PR DESCRIPTION
**Pull request by M-Lipin**
Adding IsHandleCreated checks to prevent accessing AccessibleObject properties and calls to UIAutomation which may lead to application crash/hung.

Fixes #2432 

## Proposed changes

- Adding IsHandleCreated checks to prevent accessing AccessibleObject properties for not fully initialized control.
- Fixed the reference to child list window to prevent NullReferenceException on child control recreating.

## Customer Impact

- Customers will not experience application crash/hung in some cases when ComboBox control is being reinitialized while application/form lifecycle.

## Regression? 

- Yes

## Risk

- Minimal

### Before

Exception and application hung.

`NullReferenceException` and application crash:
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Windows.Forms.ComboBox.WmParentNotify(Message& m)
   at System.Windows.Forms.ComboBox.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

### After

No exception.

## Test methodology 

- Manual testing.
- Automation tests.

## Test environment(s) 

.NET Core SDK (reflecting any global.json):
 Version:   5.0.100-alpha1-015763
 Commit:    0d0c902b77

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.100-alpha1-015763\

Host (useful for support):
  Version: 5.0.0-alpha.1.19564.1
  Commit:  c77948d92a

.NET Core SDKs installed:
  3.1.100-preview3-014645 [C:\Program Files\dotnet\sdk]
  5.0.100-alpha1-015763 [C:\Program Files\dotnet\sdk]

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2449)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.cs [hunk 1]
```diff
@@ -2805,23 +2810,19 @@ protected override void OnSelectedIndexChanged(EventArgs e)
...
-                        {
-                            accessibleObject.SetComboBoxItemFocus();
-                        }
-
-                        accessibleObject.SetComboBoxItemSelection();
+                        accessibleObject.SetComboBoxItemFocus();
                     }
+
+                    accessibleObject.SetComboBoxItemSelection();
                 }
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.cs]: ❗️  There are three levels of nesting, which is very hard to follow.
Please invert ifs/reduce nesting. For instance, add a `return` after line:2811 and remove following `else`.

Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/ComboBox.cs]: Thanks @RussKie , I've fixed this.
